### DISCUSSION TITLE
[done] When can I do standardized localization translation

### DISCUSSION THREAD

**Discussion by snmin** [totumonline/totum-ru-issues-and-discussions#101] [Есть идея, а что если...]
Hi, I see there was an update yesterday to do standardized localization translations? I can't wait to localize totum-mit.

**Comment by ttm-alexeypolunin** [comment]
I am still working on this task. I think the PR with translate pack will be on the next week

**Reply by snmin** [reply]
Hi, how is the progress of the PR with translate pack, I refresh totum-mit on github every day.

**Reply by ttm-alexeypolunin** [reply]
We are now testing this change, because it may break schemas after the update

**Reply by ttm-alexeypolunin** [reply]
Today this task move from the special branch to dev. I am sure that we will upload it to the master next week.

**Reply by ttm-alexeypolunin** [reply]
**Done from now 2.4.40.***

You have this file https://github.com/totumonline/totum-mit/blob/master/totum/moduls/install/en.json with keys for start schemas.

When you translate it as `"{{Action_available_on_viewing}}": "可在观看时采取行动",` (of course i used translator 😄 ) and create `cn.json` you can make a PR in `dev` branch with 3 lang files. After that we will add `CN` lang in installation scripts.

**Comment by snmin** [comment]
The en.json file has been translated, and I am confident that RU.php can be translated, but there are still some difficulties in translating the ru.js file. Can the ru.js file be unified like the en.json file?

**Reply by ttm-alexeypolunin** [reply]
Use this https://github.com/totumonline/totum-front/blob/dev/i18n/ru.js

**Comment by snmin** [comment]
Ok thanks, the file you sent me is very clearly typed. I am confident that I can translate it well. But it's very late today, and I need to go to translate tomorrow. At this moment, it is 1.20 am Beijing time. In addition, Chinese localization is added. In addition to these 3 files that need to be translated, what else needs to be adjusted and modified?

**Reply by ttm-alexeypolunin** [reply]
You can create repo in you account with this 3 files and send link in this discussion. We will take it and will make composed json and will add into the installation module.

**Comment by snmin** [comment]
Hello, this is my translation from the file, please use it first and make the combined json and add it to the install module. Because there is no installation and use, some translations are not very accurate. I will continue to optimize and modify the submission in the subsequent installation and use. Thanks!

snmin/totum-mit_zh: totum-mit 中文翻译 (github.com)<https://github.com/snmin/totum-mit_zh>

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

发件人: Alexey ***@***.***>
发送时间: 2022年5月7日 1:53
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
主题: Re: [totumonline/totum-ru-issues-and-discussions] When can I do standardized localization translation (Discussion #101)


You can create repo in you account with this 3 files and send link in this discussion. We will take it and will make composed json and will add into the installation module.

―

**Reply by ttm-alexeypolunin** [reply]
Great!. We will continue this work on May 11. Before that, Tanya will ask you in this tread about variants of translation for some standart `js` modules, because we saw that they have a few variations to use.

**Reply by ttm-alexeypolunin** [reply]
**Done in 2.4.41 — zh**

<img width="990" alt="Снимок экрана 2022-05-11 в 17 18 39" src="https://user-images.githubusercontent.com/55755565/167872175-564efe22-8d48-4af3-a5af-aed751aff4fe.png">

**Comment by snmin** [comment]
Thanks a lot, I'm using it now. I also found that some of the translations are not very accurate, and I will re-send them to you after I have sorted them out. Thank you once again.

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

发件人: Alexey ***@***.***>
发送时间: 2022年5月11日 22:19
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
主题: Re: [totumonline/totum-ru-issues-and-discussions] When can I do standardized localization translation (Discussion #101)


Done in 2.4.41

[Снимок экрана 2022-05-11 в 17 18 39]<https://user-images.githubusercontent.com/55755565/167872175-564efe22-8d48-4af3-a5af-aed751aff4fe.png>

―

**Comment by snmin** [comment]
I'm glad to receive your reply. I'll try localization translation now. thank you!

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

发件人: Alexey ***@***.***>
发送时间: 2022年5月6日 18:56
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
主题: Re: [totumonline/totum-ru-issues-and-discussions] When can I do standardized localization translation (Discussion #101)


Done from now 2.4.40.*

You have this file https://github.com/totumonline/totum-mit/blob/master/totum/moduls/install/en.json with keys for start schemas.

When you translate it as "{{Action_available_on_viewing}}": "可在观看时采取行动", (of course i used translator 😄 ) and create cn.json you can make a PR in dev branch with 3 lang files. After that we will add CN lang in installation scripts.
